components/brave_federated/util/linear_algebra_util.cc

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/util/linear_algebra_util.h"
+
+namespace brave_federated {
+
+Vector LinearAlgebraUtil::SubtractVector(Vector vector_1, Vector vector_2) {
+  Eigen::Map<Eigen::VectorXf> vector_1_eigen(vector_1.data(), vector_1.size());
+  Eigen::Map<Eigen::VectorXf> vector_2_eigen(vector_2.data(), vector_2.size());
+  vector_1_eigen -= vector_2_eigen;
+
+  return MatrixXfToVector(vector_1_eigen);
+}
+
+Vector LinearAlgebraUtil::MultiplyMatrixVector(Matrix matrix, Vector vector) {
+  Eigen::MatrixXf matrix_eigen = MatrixToMatrixXf(matrix);
+  Eigen::Map<Eigen::VectorXf> vector_eigen(vector.data(), vector.size());
+
+  Eigen::MatrixXf result = matrix_eigen * vector_eigen;
+  return MatrixXfToVector(result);
+}
+
+Vector LinearAlgebraUtil::AddVectorScalar(Vector vector, float scalar) {
```

Comment:
should pass by const & (see other places too)

Reply:
it won't allow me to pass it by const &, as that data is passed to the constructor of another object.